### PR TITLE
Test expressions and values via JDBC

### PR DESCRIPTION
**Pull request by kennknowles**
Adds a testing path for expressions via JDBC as well as testing that retrieving a TIMESTAMP works.

Currently, there is a failure that looks due to timezone confusion.

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

It will help us expedite review of your Pull Request if you tag someone (e.g. `@username`) to look at it.

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_GradleBuild/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_GradleBuild/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark_Gradle/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | --- | --- | --- | ---

### CODE REVIEW

**Comment by kennknowles** [comment]
@apilloud do you see any obvious error in what I did in `JdbcTest`?

File: sdks/java/extensions/sql/src/test/java/org/apache/beam/sdk/extensions/sql/impl/JdbcDriverTest.java [hunk 1]
```diff
@@ -198,6 +201,33 @@ public void testSelectsFromExistingTable() throws Exception {
...
+    ReadableInstant july1 =
+        ISODateTimeFormat.dateTimeParser().parseDateTime("2018-07-01T01:02:03Z");
+    tableProvider.addRows("test", Row.withSchema(schema).addValue(july1).build());
+
+    ResultSet selectResult =
+        connection.createStatement().executeQuery(String.format("SELECT ts FROM test"));
+    selectResult.next();
+    Timestamp ts = selectResult.getTimestamp(1);
+
+    assertThat(
```
Comment (kennknowles) [inline sdks/java/extensions/sql/src/test/java/org/apache/beam/sdk/extensions/sql/impl/JdbcDriverTest.java]: Regarding my comment: this test fails on my laptop, off by the size of the timezone. So I think there's some locale-dependent path here.

**Comment by kennknowles** [comment]
OK, this was accidentally green because Jenkins has a zero timezone offset. It would have failed for most environments, though. I added tests with zero, nonzero, and unspecified timezones, and sickbayed the ones that don't work right now.

File: sdks/java/extensions/sql/src/test/java/org/apache/beam/sdk/extensions/sql/impl/JdbcDriverTest.java [hunk 2]
```diff
@@ -198,6 +205,91 @@ public void testSelectsFromExistingTable() throws Exception {
...
+  public void testTimestampWithDefaultTimezone() throws Exception {
+    TestTableProvider tableProvider = new TestTableProvider();
+    Connection connection = JdbcDriver.connect(tableProvider);
+
+    // A table with one TIMESTAMP column
+    Schema schema = Schema.builder().addDateTimeField("ts").build();
+    connection.createStatement().executeUpdate("CREATE TABLE test (ts TIMESTAMP) TYPE 'test'");
+
+    ReadableInstant july1 =
+        ISODateTimeFormat.dateTimeParser().parseDateTime("2018-07-01T01:02:03Z");
```
Comment (apilloud) [inline sdks/java/extensions/sql/src/test/java/org/apache/beam/sdk/extensions/sql/impl/JdbcDriverTest.java]: This is taking an opinion that no timezone offset is the same as a zero offset. This is definitely a opinion but probably not one everyone will agree with. (No offset being the local timezone is also a common opinion, one that will probably make this test pass.)

File: sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/JdbcDriver.java [hunk 3]
```diff
@@ -133,7 +134,8 @@ private static void setDefault(Properties info, ConnectionProperty key, String v
     }
   }
 
-  static CalciteConnection connect(TableProvider tableProvider) {
+  @VisibleForTesting
```
[the remark this answers is not in the document]
Comment (kennknowles) [reply sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/JdbcDriver.java]: So, before I realized I could do it this way, I did it in a way that is perhaps even more faithful to what we want: I wrote a temp file and created an external table, then selected the values from it. Does that test exist elsewhere?

**Pull request merged**